lib/eco/incparser/incparser.py

```diff
@@ -496,22 +516,27 @@ def out_of_context_analysis(self, node):
               logging.debug("OOC analysis of %s failed. Error on %s.", node, temp_parser.error_nodes)
               self.error_nodes.extend(temp_parser.error_nodes)
               node.log[("left", self.prev_version)] = saved_left
-              self.isolate(node) # XXX actually need to call refine here
-              node.isolated = True
+              temp_error = node.isolated
+              # XXX has this subtree already been isolated/refined within
```

Comment:
This is again a comment for me to think a bit more about the algorithm. In theory the out-of-context analysis should try and parse the subtree and if unsuccessful isolated it. So when we reach this point it *should* already be isolated. I'll have to double check that this is true because then we can remove the `self.isolate()` call.